plutus-use-cases/test/Spec/Future.hs

```diff
@@ -159,6 +163,11 @@ settleEarly :: MonadEmulator (TraceError FutureError) m => ContractTrace FutureS
 settleEarly = do
     let
         spotPrice = Ada.lovelaceValueOf 11240
-        ov = OracleValue oracle (Ledger.Slot 25) spotPrice
+        ov = mkSignedMessage (Ledger.Slot 25) spotPrice
     callEndpoint @"settle-early" (Wallet 2) ov
     handleBlockchainEvents (Wallet 2)
+
+mkSignedMessage :: Slot -> Value -> SignedMessage (Observation Value)
+mkSignedMessage sl vl =
+    let obs = Observation{obsValue=vl, obsSlot=sl} in
+    Oracle.signMessage obs (fst oracleKeys)
```

Comment:
Make it polymorphic over the type of `obsValue` and this could go in the `Oracle` module.